src/plugin/luapi_application.h

```diff
@@ -214,7 +237,58 @@ static int applib_uiActionSelected(lua_State* L) {
 }
 
 /**
- * Change page background
+ * Execute action from sidebar menu
+ *
+ * Example: app.sidebar_Action("MOVE_DOWN")
+ * moves down the current page
+ */
+static int applib_sidebarAction(lua_State* L) {
+    // Connect the context menu actions
+    const std::map<std::string, SidebarActions> actionMap = {
+            {"COPY", SIDEBAR_ACTION_COPY},
+            {"DELETE", SIDEBAR_ACTION_DELETE},
+            {"MOVE_UP", SIDEBAR_ACTION_MOVE_UP},
+            {"MOVE_DOWN", SIDEBAR_ACTION_MOVE_DOWN},
+            {"NEW_BEFORE", SIDEBAR_ACTION_NEW_BEFORE},
+            {"NEW_AFTER", SIDEBAR_ACTION_NEW_AFTER},
+    };
+    const char* actionStr = luaL_checkstring(L, 1);
+    if (actionStr == nullptr) {
+        luaL_error(L, "Missing action!");
+    }
+    auto pos = actionMap.find(actionStr);
+    if (pos == actionMap.end()) {
+        luaL_error(L, "Action unknown!");
+    }
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    SidebarToolbar* toolbar = plugin->getControl()->getSidebar()->getToolbar();
+    toolbar->runAction(pos->second);
+
+    return 1;
+}
+
+/**
+ * Execute action from layer controller
+ *
+ * Example: app.layerAction("ACTION_DELETE_LAYER")
+ * deletes the current layer
+ */
+static int applib_layerAction(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+
+    ActionType action = ActionType_fromString(luaL_checkstring(L, 1));
```

Comment:
This should warn if the action is `ACTION_NONE` (i.e. invalid action).

Reply:
If the action is invalid then `ActionType_fromString` will show an error message. It won't use `ACTION_NONE` as return value. I could check manually if `action` is one of the Action types handled in `LayerController::actionPerformed` (those are `ACTION_NEW_LAYER` , `ACTION_DELETE_LAYER`, `ACTION_GOTO_PREVIOUS_LAYER` and `ACTION_GOTO_TOP_LAYER`), but if more actions are added or the actions are renamed, this check will have to be updated again. So what should I do?

Reply:
Ah, I didn't see that error message.